src/Validation/Requires.cs

```diff
@@ -457,6 +457,79 @@ public static void That([DoesNotReturnIf(false)] bool condition, string? paramet
             }
         }
 
+        /// <summary>
+        /// Throws an exception if the specified parameter's value is not null
```

Comment:
You claim that you only throw an exception if the parameter's value is "not null". But in your latest iteration you now throw even if it _is_ null. So this bears some fixing.
Also you can shorten "parameter's value" to just "argument". 

Same feedback on your other overload.

Reply:
I'll update the comments.